runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -3292,23 +3293,38 @@ TR_RelocationRecordProfiledInlinedMethod::preparePrivateData(TR_RelocationRuntim
       }
    else
       {
-      J9ROMClass *inlinedCodeRomClass = reloRuntime->fej9()->sharedCache()->romClassFromOffsetInSharedCache(romClassOffsetInSharedCache(reloTarget));
+      auto sharedCache = reloRuntime->fej9()->sharedCache();
+      uintptr_t romClassOffset = romClassOffsetInSharedCache(reloTarget);
+      J9ROMClass *inlinedCodeRomClass = sharedCache->romClassFromOffsetInSharedCache(romClassOffset);
       J9UTF8 *inlinedCodeClassName = J9ROMCLASS_CLASSNAME(inlinedCodeRomClass);
-      RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: inlinedCodeRomClass %p %.*s\n", inlinedCodeRomClass, J9UTF8_LENGTH(inlinedCodeClassName), J9UTF8_DATA(inlinedCodeClassName));
+      RELO_LOG(reloRuntime->reloLogger(), 6, "\tpreparePrivateData: inlinedCodeRomClass %p %.*s\n", inlinedCodeRomClass,
+               J9UTF8_LENGTH(inlinedCodeClassName), J9UTF8_DATA(inlinedCodeClassName));
 
-      void *classChainIdentifyingLoader = reloRuntime->fej9()->sharedCache()->pointerFromOffsetInSharedCache(classChainIdentifyingLoaderOffsetInSharedCache(reloTarget));
-      RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: classChainIdentifyingLoader %p\n", classChainIdentifyingLoader);
-      J9ClassLoader *classLoader = (J9ClassLoader *) reloRuntime->fej9()->sharedCache()->lookupClassLoaderAssociatedWithClassChain(classChainIdentifyingLoader);
-
-      RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: classLoader %p\n", classLoader);
+      void *classChainIdentifyingLoader = sharedCache->pointerFromOffsetInSharedCache(
+                                          classChainIdentifyingLoaderOffsetInSharedCache(reloTarget));
+      RELO_LOG(reloRuntime->reloLogger(), 6, "\tpreparePrivateData: classChainIdentifyingLoader %p\n",
+               classChainIdentifyingLoader);
+      J9ClassLoader *classLoader = (J9ClassLoader *)sharedCache->persistentClassLoaderTable()
```

Comment:
The old code was calling`reloRuntime->fej9()->sharedCache()->lookupClassLoaderAssociatedWithClassChain(classChainIdentifyingLoader);`

`lookupClassLoaderAssociatedWithClassChain()` is virtual and has two implementations
```
void *
TR_J9SharedCache::lookupClassLoaderAssociatedWithClassChain(void *chainData)
   {
   return persistentClassLoaderTable()->lookupClassLoaderAssociatedWithClassChain(chainData);
   }
   ```
   and 
   ```
void *
TR_J9DeserializerSharedCache::lookupClassLoaderAssociatedWithClassChain(void *chainData)
   {
   // We return chainData directly because the only thing this function can be called on is the result of
   // TR_J9DeserializerSharedCache::pointerFromOffsetInSharedCache(), and that will return a (J9ClassLoader *)
   // directly when given a class loader offset.
   return chainData;
   }
   ```
   The new code calls the first implementation indiscriminately. We should use the virtual call like in the original